File: python_files/raylib/center_and_draw_text.py
import pyray as p


def center_and_draw_text(
    text: str,
    font_size: int,
    layout_rect: p.Rectangle,
    color: p.Color = p.BLACK,
) -> None:
    text_width: int = p.measure_text(text, font_size)
    text_x: int = int(layout_rect.x + layout_rect.width / 2 - text_width / 2)
    text_y: int = int(layout_rect.y + layout_rect.height / 2 - font_size / 2)

    p.draw_text(text, text_x, text_y, font_size, color)

File: cpp_files/raylib/center_and_draw_text.cpp
#include <raylib.h>

void
center_and_draw_text(const char* text, int font_size, Rectangle layout_rect, Color color = BLACK)
{
    int text_width = MeasureText(text, font_size);
    int text_x = layout_rect.x + layout_rect.width / 2 - text_width / 2;
    int text_y = layout_rect.y + layout_rect.height / 2 - font_size / 2;

    DrawText(text, text_x, text_y, font_size, color);
}